### PR TITLE
feat: add finalizer to snapshots

### PR DESCRIPTION
**Pull request by Skarlso**
The finalizer should delete any cached data that is assigned to the snaphost.

### CODE REVIEW

**Comment by Skarlso** [comment]
Uh, I'm doing something silly... This is isn't working yet. :D

**Comment by Skarlso** [comment]
Ok, Figured it out. I can only delete via digest. Makes sense. :)

**Comment by Skarlso** [comment]
Uh, this thing isn't working sadly. I will have to investigate it more.

**Comment by Skarlso** [comment]
Finally, a way forward. Get the manifest's digest and delete that. That should get rid of all the tags.

File: controllers/snapshot_controller.go [hunk 1]
```diff
@@ -140,3 +163,26 @@ func (r *SnapshotReconciler) Reconcile(ctx context.Context, req ctrl.Request) (c
...
+		return fmt.Errorf("failed to reconcile delete: %w", err)
+	}
+
+	name, err := ocm.ConstructRepositoryName(obj.Spec.Identity)
+	if err != nil {
+		return fmt.Errorf("failed to construct name: %w", err)
+	}
+
+	if err := r.Cache.DeleteData(ctx, name, obj.Spec.Tag); err != nil {
+		if !strings.Contains(err.Error(), "MANIFEST_UNKNOWN") {
```
Comment (phoban01) [inline controllers/snapshot_controller.go:180]: Does this not return status code we could use instead of the string?

**Review by phoban01 — APPROVED** [review]
Thanks for all the work on this. This was definitely an iceberg issue to tackel!

File: controllers/snapshot_controller.go (hunk 1 above)
Comment (Skarlso) [reply controllers/snapshot_controller.go:180]: It's a bit ugly, because the manifest unknown will be in the diagnostics list. Pf.

Comment (Skarlso) [reply controllers/snapshot_controller.go]: Done. Verifying now if Unknown Manifest is in the list of diagnostic errors. I can't do better. :D

**Pull request merged**